src/components/QueryEditor.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { ChangeEvent, useRef } from 'react';
+import { InlineField, Input } from '@grafana/ui';
+import { QueryEditorProps } from '@grafana/data';
+import { CriblDataSource } from '../datasource';
+import { CriblDataSourceOptions, CriblQuery } from '../types';
+import { debounce } from 'lodash';
+
+type Props = QueryEditorProps<CriblDataSource, CriblQuery, CriblDataSourceOptions>;
+
+export function QueryEditor({ query, onChange, onRunQuery }: Props) {
+  const debouncedOnRunQuery = useRef(debounce(onRunQuery, 750)).current;
+
+  const onSavedQueryIdChange = (event: ChangeEvent<HTMLInputElement>) => {
+    onChange({ ...query, savedQueryId: event.target.value });
+    debouncedOnRunQuery();
+  };
+
+  const onMaxResultsChange = (event: ChangeEvent<HTMLInputElement>) => {
+    onChange({ ...query, maxResults: +event.target.value });
+    debouncedOnRunQuery();
+  };
+
+  const { savedQueryId, maxResults } = query;
+
+  return (
+    <div className="gf-form">
```

Comment:
Any chance that we could add test/verify here?

- is the saved-queryi-d valid?
- do I have access (ACL) to the saved query?

Reply:
Yes and no.  Yes in that we can at least ensure `savedQueryId` isn't _obviously invalid_ (i.e. matches the expected regex), which I'll add.  But no in that the `/search/query` API should take care of enforcing ACL without requiring a separate check for that.